components/gateway/internal/gqlschema/schema.graphql

```diff
@@ -0,0 +1,412 @@
+    labels: Labels
+    annotations: Annotations
+    webhooks: [ApplicationWebhookInput]
+    healthCheckURL: String
+    apis: [APIInput!]
+    events: [EventInput!]
+}
+
+# Runtime Input
+
+input RuntimeInput {
+    name: String!
+    description: String
+    labels: Labels
+    annotations: Annotations
+    agentCredential: CredentialInput!
+}
+
+# FetchRequest Input
+
+input FetchRequestInput {
+    url: String
+    credential: CredentialInput
+}
+
+# Webhook Input
+
+input ApplicationWebhookInput {
+    type: ApplicationWebhookType!
+    url: String!
+    credential: CredentialInput
+}
+
+# API Input
+
+input APIInput {
+    targetURL: String!
+    credential: CredentialInput
+    injectHeaders: HttpHeaders
+    injectQueryParams: QueryParams
+    documentations: [DocumentationInput!]
+    spec: APISpecInput
+
+}
+
+input APISpecInput {
+    type: APISpecType!
+    data: Clob
+    fetchRequest: FetchRequestInput
+}
+
+# Event Input
+
+input EventInput {
+    type: EventSpecType!
+    data: Clob
+    fetchRequest: FetchRequestInput
+    documentations: [DocumentationInput!]
+}
+
+# Documentation Input
+
+input DocumentationInput {
+    title: String!
+    displayName: String!
+    description: String!
+    format: DocumentationFormat!
+    kind: String
+    data: Clob
+    fetchRequest: FetchRequestInput
+}
+
+
+# Credential Input
+
+input CredentialInput {
+    data: CredentialDataInput!
+    requestAuth: CredentialRequestAuthInput
+}
+
+input CredentialRequestAuthInput {
+    type: CredentialRequestAuthType!
+    csrf: CSRFTokenCredentialRequestAuthInput
+}
+
+input CSRFTokenCredentialRequestAuthInput {
+    token: String!
+}
+
+input CredentialDataInput {
+    type: CredentialDataType!
+    basic: BasicCredentialDataInput
+    oauth: OAuthCredentialDataInput
+}
+
+input OAuthCredentialDataInput {
+    clientId: ID!
+    clientSecret: String!
+    url: String!
+}
+
+input BasicCredentialDataInput {
+    username: String!
+    password: String!
+}
+
+# HealthCheck Input
+
+
+enum FilterOperator {
+    ALL, ANY
+}
+
+input LabelFilter {
+    label: String!
+    values: [String!]!
+    operator: FilterOperator
+}
+
+type Query {
+    applications(filer: [LabelFilter!]): [Application!]!
+    application(id: ID!): Application
+
+    runtimes(filter: [LabelFilter!]): [Runtime!]!
+    runtime(id: ID!): Runtime
+
+    healthChecks(types: [HealthCheckType], originID: ID): [HealthCheck!]!
+    labels(key: String): [String!]!
+
+}
+
+type Mutation {
+    # Application
+    createApplication(in: ApplicationInput!): Application
+    updateApplication(id: ID!, in: ApplicationInput!): Application
+    deleteApplication(id: ID!): Application
+
+    createApplicationLabel(applicationID: ID!, label: String!, values: [String!]!): [String!]!
+    deleteApplicationLabel(applicationID: ID!, label: String!, values: [String!]!): [String!]!
+
+    createApplicationAnnotation(applicationID: ID!, annotation: String!, value: String!): String!
```

Comment:
Should the returned type be not nullable? Also, are we returning only value (applies also to label mutations)?